### PR TITLE
SRP recent file share API change

### PR DESCRIPTION
**Pull request by zfchen95**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by zfchen95** [comment]
Merge changes in PR https://github.com/Azure/azure-rest-api-specs/pull/7815 and https://github.com/Azure/azure-rest-api-specs/pull/7988 to one place

**Comment by zfchen95** [comment]
There is a required check that is failed because of adding new properties to share. The properties are already added and it did not cause breaking from server side. Is there a way to bypass it?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json [hunk 1]
```diff
@@ -524,6 +629,86 @@
           "minimum": 1,
           "maximum": 102400,
           "description": "The maximum size of the share, in gigabytes. Must be greater than 0, and less than or equal to 5TB (5120). For Large File Shares, the maximum size is 102400."
+        },
+        "enabledProtocols": {
+          "type": "string",
+          "enum": [
+            "SMB",
+            "NFS"
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:637]: As we get on the last week, NFS feature has server issue, and seems won't be ready before June. (correct me if not)
@zfchen95  Would you please check with Will that he really want to open NFS in swagger now? If the change is in swagger, means it's open in rest API and all SDKs. Then customer might use it with SDK, and if feature is not ready, we might will get many customer report issues.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json [hunk 2]
```diff
@@ -524,6 +629,86 @@
...
+        },
+        "accessTierStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Indicates if there is a pending transition for access tier."
+        },
+        "shareUsageBytes": {
+          "type": "integer",
+          "readOnly": true,
+          "description": "The approximate size of the data stored on the share. Note that this value may not include all recently created or recently resized files."
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:711]: @zfchen95 
As we get on the last meeting, we will have feature "Share status" GA soon. 
But I don't see it in this PR, When will add it?

Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:711]: There is no share status. I believe you are referring to share tier

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json (hunk 1 above)
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:637]: I forwarded you an email from Ran. We still plan to public preview the feature in limited regions.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:637]: See, thanks!

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json (hunk 2 above)
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:711]: OK, thanks!

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json [hunk 3]
```diff
@@ -524,6 +629,86 @@
...
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The deleted time if the share was deleted."
+        },
+        "remainingRetentionDays": {
+          "type": "integer",
+          "readOnly": true,
+          "description": "Remaining retention days for share that was soft deleted."
+        },
+        "accessTier": {
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:683]: What's the value user can set when update/create share?
I think not all value can set set by user, like user should can't set it to premium from other value.
Would you please add it to description to make the usage more clear?

Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:683]: Edited the description.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:683]: hot and cool can only be set to GPV2, or to all accounts?

Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:683]: GPv2 only. We don't support GPv1. FileStorage is Premium only.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:683]: Would you please indicate it in the description?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/DeletedFileSharesList.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    "resourceGroupName": "res9290",
+    "accountName": "sto1590",
+    "$expand": "deleted",
+    "api-version": "2019-06-01",
+    "monitor": "true"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/DeletedFileSharesList.json:13]: For the response of list operation,  should we return an array directly not a json with array in "value"?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    "shareName": "share1249",
+    "deletedShare": {
+      "deletedShareName": "share1249",
+      "deletedShareVersion": "1234567890"
+    },
+    "api-version": "2019-06-01",
+    "monitor": "true"
+  },
+  "responses": {
+    "200": {}
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json:15]: Please add response in examples to help us understand the output format.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json [hunk 6]
```diff
@@ -410,6 +448,56 @@
...
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}/restore": {
+      "post": {
+        "tags": [
+          "FileShares"
+        ],
+        "operationId": "FileShares_Restore",
+        "x-ms-examples": {
+          "PutShares": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:459]: RestoreShares maybe better here.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json [hunk 7]
```diff
@@ -350,6 +375,19 @@
...
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "enum": [
+              "stats"
+            ],
+            "x-ms-enum": {
+              "name": "GetShareExpand"
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:388]: kind reminder: Do you want to model as String here?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/DeletedFileSharesList.json (hunk 4 above)
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/DeletedFileSharesList.json:13]: It is by design. We may have nextLink alongside with value. We return limited number of shares, depends on the page size.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json (hunk 5 above)
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json:15]: ah.. the response body will be empty. The status code indicates it is succeeded

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json (hunk 6 above)
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json:459]: you are right! good catch here. I might copy it from PutShares and forgot to change it.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json (hunk 7 above)
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json]: I have no strong preference on modelAsString. Will set it to false to match StorageAccountExpand, which is modeled as enum.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json (hunk 5 above)
Comment (Juliehzl) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/FileSharesRestore.json:15]: I see.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**